scripts/pylib/twister/twisterlib/runner.py

```diff
@@ -87,37 +87,65 @@ def __init__(self, total=0):
         # updated by report_out()
         self._skipped_runtime = Value('i', 0)
 
-        # staic filtered at yaml parsing time
+        # static filtered at yaml parsing time
         # updated by update_counting_before_pipeline()
         self._skipped_filter = Value('i', 0)
 
-        # updated by update_counting_before_pipeline() and report_out()
-        self._skipped_cases = Value('i', 0)
-
         # updated by report_out() in pipeline
         self._error = Value('i', 0)
         self._failed = Value('i', 0)
 
         # initialized to number of test instances
         self._total = Value('i', total)
 
+        #######################################
+        # TestCase counters for all instances #
+        #######################################
         # updated in report_out
         self._cases = Value('i', 0)
+
+        # updated by update_counting_before_pipeline() and report_out()
+        self._skipped_cases = Value('i', 0)
+        self._filtered_cases = Value('i', 0)
+
+        # updated by report_out() in pipeline
+        self._passed_cases = Value('i', 0)
+        self._failed_cases = Value('i', 0)
+        self._error_cases = Value('i', 0)
+        self._blocked_cases = Value('i', 0)
+
+        # Incorrect statuses
+        self._none_cases = Value('i', 0)
+        self._started_cases = Value('i', 0)
+
+
         self.lock = Lock()
 
     def summary(self):
         print("--------------------------------")
         print(f"Total test suites: {self.total}") # actually test instances
-        print(f"Total test cases: {self.cases}")
-        print(f"Executed test cases: {self.cases - self.skipped_cases}")
-        print(f"Skipped test cases: {self.skipped_cases}")
-        print(f"Completed test suites: {self.done}")
+        print(f"Added to pipeline test suites: {self.done}")
```

Comment:
hmm, I might be lost in the example from PR description, but the `New style` is still something I need to translate in my mind (ordered more like as it happened):
```
Test Suites:
 scheduled(32) = filtered(17) + built(15)
 executed(15) = filtered(1) + errored(1) + failed(1) + passed(12)

Test Cases in Suites - built(148):
 not executed (30) = filtered(28) + skipped(2)
 executed (118) = no status(6) +  blocked(1) + errored(0) + failed(1) + passed(110)
```

Reply:
I've taken a page out of Linux `tree` command and designed this summary:
```
--------------------------------------------------
Total test suites: 32
Processed test suites: 32
├─ Skipped test suites (static filter): 17
└─ Completed test suites: 15
   ├─ Passing test suites: 12
   ├─ Failing test suites: 1
   ├─ Errors in test suites: 1
   └─ Skipped test suites (at runtime): 1

Skipped test suites: 18
├─ Skipped test suites (at runtime): 1
└─ Skipped test suites (static filter): 17
----------------------      ----------------------
Total test cases: 148
├─ Filtered test cases: 28
├─ Skipped test cases: 2
└─ Executed test cases: 118
   ├─ Failed test cases: 2
   ├─ Error test cases: 0
   ├─ Blocked test cases: 1
   ├─ Passed test cases: 110
   └─ Test cases without a status: 6
--------------------------------------------------
```

It duplicates some information (test suite skips), but I think it is worth it for the clarity of the connections between the statistics.

Reply:
Just few suggestions:
* let's always think this output will be parsed by someone with a script using regular expressions,  etc.
* use same words in same form (like 'Passed', not 'Passing').
* keep rows ordered in sequence of state transitions.

```
Total Skipped test Suites: 18
├─ Skipped test Suites (static filter): 17
└─ Skipped test Suites (at runtime): 1
```

And naive questions:
* Why ` Executed test cases: 118` don't match to the sum at its sub-items ?

* what to do with this ?
```
Test cases without a status: 6
```

Reply:
In order:

- OK; new output should be easily processed by relying on common breaking symbols - there is always colon, then whitespace, then number and the first word of each line is unique if grouped by tree.
- OK; Naming is more consistent and a bit shorter now.
- OK; now the order is roughly conforming to the processing order.
- This is because we have retries. The failed instance is retried and its cases counted twice. I should reset it like the test suites are when a retry is attempted. From my code analysis, a retry assumes the previous run null and void and should not be reported in the summary. I've changed that so failed instance testcases are not double-counted
- This is an indication that the current status system has its problems. `NONE` and `STARTED` statuses are not proper terminal statuses. However, this change did not cause them, but merely extracted them so they can be viewed by the user. I have added an additional line shortly explaining that to the end user, easily discarded by an automatic user.

With those comments in mind and with formatting changes (right-alignment for numbers, common column length), here is a new design:

```
--------------------------------------------------
Total test suites:     32
Processed test suites: 32
├─ Skipped test suites (static filter): 17
└─ Completed test suites:               15
   ├─ Skipped test suites (at runtime):     1
   ├─ Passed test suites:                  12
   ├─ Failed test suites:                   1
   └─ Errors in test suites:                1

Skipped test suites: 18
├─ Skipped test suites (static filter): 17
└─ Skipped test suites (at runtime):     1
----------------------      ----------------------
Total test cases: 148
├─ Filtered test cases:  28
├─ Skipped test cases:    2
└─ Executed test cases: 118
   ├─ Passed test cases:       110
   ├─ Blocked test cases:        1
   ├─ Failed test cases:         1
   ├─ Errors in test cases:      0
   ├──── The following test case statuses should not appear in a proper execution ───
   └─ Statusless test cases:     6
--------------------------------------------------
```